servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -1138,9 +1138,12 @@ public void executePutPolicyVersion(ResourceContext ctx, String domainName, Stri
 
                 // add domain change event
                 addDomainChangeMessage(ctx, domainName, policyName, DomainChangeMessage.ObjectType.POLICY);
-                
-                return;
 
+                if (returnObj == Boolean.TRUE) {
```

Comment:
consider combining these next 5 lines into a single step:

return (returnObj == Boolean.TRUE) ? getPolicy(con, domainName, policyName, version) : null;

(same comment applies to all other sections in the code).